lib/irb/completion.rb

```diff
@@ -453,16 +453,30 @@ def retrieve_completion_data(input, bind:, doc_namespace:)
         else
           candidates = (bind.eval_methods | bind.eval_private_methods | bind.local_variables | bind.eval_instance_variables | bind.eval_class_constants).collect{|m| m.to_s}
           candidates |= RubyLex::RESERVED_WORDS.map(&:to_s)
-          candidates.grep(/^#{Regexp.quote(input)}/).sort
+          safe_grep(candidates, /^#{Regexp.quote(input)}/).sort
         end
       end
     end
 
     # Set of available operators in Ruby
     Operators = %w[% & * ** + - / < << <= <=> == === =~ > >= >> [] []= ^ ! != !~]
 
+    def safe_grep(candidates, pattern)
+      target_encoding = Encoding.default_external
+      candidates.filter_map do |candidate|
+        next unless candidate
```

Comment:
I think this part is not needed because all candidate passed to `safe_grep` are converted to string with `.collect{|m| m.to_s}`

Reply:
I was surprised as well but without this  the tests fail on TruffleRuby (compare the builds on the two commits in this PR).

It's working on every other platform though. 🤔
Or was I missing a bit here or misunderstanding something?

/CC @eregon